versioned/storage/cache/src/main/java/org/projectnessie/versioned/storage/cache/CaffeineCacheBackend.java

```diff
@@ -31,24 +36,30 @@
 abstract class CaffeineCacheBackend implements CacheBackend {
 
   public static final int JAVA_OBJ_HEADER = 32;
+  public static final String CACHE_NAME = "nessie-objects";
 
   static ImmutableCaffeineCacheBackend.Builder builder() {
     return ImmutableCaffeineCacheBackend.builder();
   }
 
   abstract long capacity();
 
+  @Nullable
+  @jakarta.annotation.Nullable
+  abstract MeterRegistry meterRegistry();
+
   @Value.Derived
   Cache<CacheKey, byte[]> cache() {
-    // IMPORTANT!
-    // When changing the configuration of the Caffeine cache, make sure to run the
-    // _native_ Quarkus tests and adopt the `@ReflectionConfig` in
-    // org.projectnessie.quarkus.providers.PersistProvider.
-    return Caffeine.newBuilder()
-        .maximumWeight(capacity())
-        .recordStats()
-        .weigher(this::weigher)
-        .build();
+    Caffeine<CacheKey, byte[]> cacheBuilder =
+        Caffeine.newBuilder().maximumWeight(capacity() * 1024L * 1024L).weigher(this::weigher);
+    MeterRegistry meterRegistry = meterRegistry();
+    if (meterRegistry != null) {
+      cacheBuilder.recordStats(() -> new CaffeineStatsCounter(meterRegistry, CACHE_NAME));
+      meterRegistry
+          .counter("cache_capacity_mb", singletonList(Tag.of("cache", CACHE_NAME)))
```

Comment:
I would rather describe this as a `gauge`.
```suggestion
      meterRegistry.gauge("cache_capacity_mb", singletonList(Tag.of("cache", CACHE_NAME)), capacity());
```

Reply:
Nope - a gauge's reset (resulting in `NaN`), a counter is not.

Reply:
You need to store a reference to the capacity, otherwise yes, it will be garbage-collected.